include/irr/asset/format/encodePixels.h

```diff
@@ -30,18 +30,18 @@ namespace irr { namespace video
 			return -1; // handle an error
 	}
 
-	template<typename T, typename U>
-	inline void clampVariableProperly(T& variableToAssignClampingTo, const U& variableToClamp, bool isItNormalizedVariable = true)
+	template<typename T>
+	inline void clampVariableProperly(T& variableToAssignClampingTo, const double& variableToClamp, const asset::E_FORMAT &format)
 	{
 		const auto& max = getRangeValueOfVariable<T>(true);
```

Comment:
shouldn't this all be constexpr + constexpr function instead of reference?

Reply:
to fix, sure